src/main/java/io/snyk/jenkins/tools/SnykInstaller.java

```diff
@@ -121,12 +121,12 @@ private FilePath installSnykAsNpmPackage(FilePath expected, Node node, TaskListe
       expected.child(TIMESTAMP_FILE).write(valueOf(Instant.now().toEpochMilli()), UTF_8.name());
     } catch (Exception ex) {
       log.getLogger().println("Snyk Security tool could not installed: " + ex.getMessage());
-      LOG.error("Could not install Snyk as NPM package", ex);
+      throw new ToolDetectionException("Could not install Snyk CLI with npm", ex);
     }
     return expected;
   }
 
-  private FilePath installSnykAsSingleBinary(FilePath expected, Node node, TaskListener log) throws IOException, InterruptedException {
+  private FilePath installSnykAsSingleBinary(FilePath expected, Node node, TaskListener log) throws IOException, InterruptedException, ToolDetectionException {
```

Comment:
@maxjeffos, I think there is no need to add ToolDetectionException in method signature, because it extends IOException and we already have it.

Reply:
Ok, makes sense - will remove it.
Thanks @pavel-snyk